sympy/functions/combinatorial/tests/test_comb_factorials.py

```diff
@@ -334,6 +335,14 @@ def test_binomial():
 
     assert binomial(a, b).is_nonnegative is True
 
+    # issue #14625
+    assert binomial(a, a) == 1
+    assert binomial(a, a - 1) == a
+    assert isinstance(binomial(u, u), binomial)
+    assert isinstance(binomial(u, u - 1), binomial)
```

Comment:
Is binomial(x, x - 1) = x and binomial(x, x) = 1 if x.is_integer is False?

Reply:
I think so, by the Symmetry Identity `C(n, k) = C(n, n - k)`, though we should be careful as the recent behaviour change for binomial gave preference to Pascal's Identity over this (for negative k).
Although in this case it holds since n is non-negative (and the conditions added also ensure k is appropriate). The only special case being `n=0` when `k` might be transformed to `-1`, but `C(0, -1) = C(0, 1) = 0`.

Reply:
I ask because this branch leaves these unevaluated when x.is_integer is False.
```
>>> b(x,x),b(x,x-1)
(binomial(x, x), binomial(x, x - 1))
>>> b(pi,pi).n(),b(pi,pi-1).n()
(1.00000000000000, 3.14159265358979)
>>> b(-x,-x),b(-x,-x-1)
(binomial(-x, -x), binomial(-x, -x - 1))
>>> b(-pi,-pi).n(),b(-pi,-pi-1).n()
(1.00000000000000, -3.14159265358979)
```